### ISSUE TITLE
[quant][pt2] Default batchnorm aten op has poor numerics during QAT

### ISSUE THREAD

**Issue by andrewor14** [pytorch/pytorch#111384]
If a model contains batch norm, exporting `model.cpu()` gives `torch.ops.aten._native_batch_norm_legit.default` while exporting `model.cuda()` gives `torch.ops.aten.cudnn_batch_norm.default`. Quick experimentation revealed that these two ops have significantly different numerics. From running PT2 QAT on MobileNetV2, we found that these numerical differences are magnified through many back-to-back `[conv - bn]` or `[conv - bn - relu]` patterns, and training with `cudnn_batch_norm` gave significantly better accuracies.

For the short term, we recommend users of PT2 QAT to either export with `model.cuda()` from the start, or manually swap the batch norm op from `_native_batch_norm_legit` to `cudnn_batch_norm` before training. See this [tutorial](https://pytorch.org/tutorials/prototype/pt2e_quant_qat.html#prepare-the-model-for-quantization-aware-training) for more detail.

For the longer term, however, we wish to fix this so that the user does not have to do this manual swapping. We can achieve this by consolidating the batch norm ops such that exporting the model always gives the same op, regardless whether it is on CPU or cuda, and that op will automatically pick the right kernel to run on. See this relevant [comment](https://github.com/pytorch/pytorch/pull/108809#issuecomment-1724566234).
